### DISCUSSION TITLE
Is there any way that I can export all informations of edge ids from all tiles?

### DISCUSSION THREAD

**Discussion by mushabtinumbang** [valhalla/valhalla#4713] [Q&A] [answered]
Hi, I hope you are all having a good day. I am wondering if there is a way to obtain a list of all possible edge IDs, their geometries, and their osm_way_ids for every tile that we have built. To provide more context, I am not referring to a list of edge IDs from the MM result (trace_attribute options). Instead, I want to generate information for every edge ID based on the osm.pbf and the tiles that Valhalla has built. Thanks!

**Comment by kevinkreiser** [comment]
there is an executable called valhalla_export_edges whoch does this. you might have to modify it a little to get all the fields you want but it should be trivial modifications

**Reply by mushabtinumbang** [reply]
i have tried to modify the code a little. But it seems like i can't get the actual global value of edge_id, is that true? I have also compared the edge_id's result to the edge_id's passed by mapmatching results, and i think they differ. this is how i define my edge_id.

```
edge_t edge{tile_count_pair.first, tile->directededge(i)};
edge.i.set_id(i);
  
edge_set.set(tile_count_pair.second + i);
++set;
uint32_t edge_id = edge.i.id();
```

I think it's obvious because the edge_id is basically just the i in the for loop. Is there any way that I can get the actual edge_id's? Thanks

**Reply by kevinkreiser** [reply]
first, that executable tries to put together as many edges as possible into a long chain of them. if you want a row for every edge you need to stop it from doing that which is a trivial feature we could add.

to get the edge id you want `edge.i` the full 64bit value is the "global edge id".

**Reply by mushabtinumbang** [reply]
superb! thanks for the answer, kevin! really appreciate it.

**Reply by mushabtinumbang** [reply]
hi kevin, im just wondering, can we extract the edges based on certain region? say that I have built the tiles from 2 osm.pbf e.g Bayern and Hamburg. How can i extract all of the edges only in Bayern? Thanks

**Reply by kevinkreiser** [reply, marked as answer]
did you build them in the same tileset or separate? if they are separate just run the application with whichever tileset you want to get edges from. if they are combined into a single tileset then you'll need to add code to decide if an edge is within your target area or not. to do that there are varying degrees of exactness. you could simply check if the begin or end node lat lon is in your lat lon bounding box, or you could check if any shape point on the edge is within your lat lon boundign box or you could actually intersect each segment of the shape with the bounding boxes edges or you could do that but take care that they are actually geodesics so you need to intersect arcs rather than line segments. like i said lots of options but varying degrees of "accuracy".

**Comment by asongtoruin** [comment]
Apologies for reviving an old thread, but I'm also interested in doing something similar to this - I'd like to export each edge from an entire Valhalla network including the internal ID, though I lack the C++ knowledge to implement the suggestions above. I notice it was mentioned that the individual edge export (and I assume, adding the IDs) might be a trivial feature, but I'm guessing this hasn't been added yet?

**Reply by kevinkreiser** [reply]
we have an executable called `valhalla_export_edges` it will export edges names and shape however what it tries to do is connect all the edges into the longest possible section where the names are the same which means you'd have more than one edge per row of output from this application. you could however hack it to not do that. you'd need to modify the `next` function to always return `{}` meaning no more edges to contract with the start edge. you'd then also need to modify the printing bit to also print out the edge ids which for this is simply `edge.i`. seems like something you could hack up pretty quickly for your needs
